src/include/migraphx/op/isnan.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef MIGRAPHX_GUARD_OPERATORS_ISNAN_HPP
+#define MIGRAPHX_GUARD_OPERATORS_ISNAN_HPP
+
+#include <migraphx/op/unary.hpp>
+#include <migraphx/config.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+struct isnan : unary<isnan>
+{
+    std::string point_function() const { return "isnan"; }
+    auto apply() const
+    {
+        return [](auto x) { return std::isnan(x); };
+    }
+
+    std::string name() const { return "isnan"; }
+
+    shape compute_shape(std::vector<shape> inputs) const
```

Comment:
This is already handled by the `unary<isnan>` base class.

Reply:
The compute_shape method in unary.hpp was returning a shape with the same type as the input, but ONNX specifies that the output is bool.

Reply:
Then call `unary<isnan>` first and then change the type , but we need to add a `with_type` method to the shape class.

Reply:
Well I added `with_type` in #1102.

Reply:
I pulled the branch, looks to work by hiding and calling the base class.